src/flatten.jl

```diff
@@ -74,26 +78,26 @@ function flatten(::Type{T}, x::Tuple) where {T<:Real}
     x_vecs, x_backs = first.(x_vecs_and_backs), last.(x_vecs_and_backs)
     lengths = map(length, x_vecs)
     sz = _cumsum(lengths)
-    function unflatten_to_Tuple(v::Vector{T})
+    function unflatten_to_Tuple(v::AbstractVector{<:Real})
         map(x_backs, lengths, sz) do x_back, l, s
             return x_back(v[(s - l + 1):s])
```

Comment:
It would be amazing if we can somehow find a way to make a `@view` here, so we do not generate a new vector for each argument in the tuple. The problem is that if we have to call `NamedTuple{names}(v_vec_vec)` instead of `typeof(x)(v_vec_vec)` in the NamedTuple dispatch below, we will get back different types for everything bar scalar parameter

Reply:
Hmm I agree that that would be nice. Would you mind opening a separate issue to discuss further? I'd rather keep it out of scope for this PR.